src/api/db/connect.ts

```diff
@@ -1,34 +1,32 @@
 import mongoose from 'mongoose';
 import { InternalServerError } from '../errors.js';
+import { Config } from '../../config/config.js';
 
 // TODO: consider using multiple connections (one per model)
 // to reduce risk of slow trains
 // https://mongoosejs.com/docs/connections.html#multiple_connections
 // https://thecodebarbarian.com/slow-trains-in-mongodb-and-nodejs
 
-let cachedConnectionPromise = null;
+let client: typeof mongoose | null = null;
 
-async function connectToDatabase(config) {
-  if (!cachedConnectionPromise) {
+async function connectToDatabase(config: Config) {
```

Comment:
any reason we're not specify the return type here?

Reply:
Not really.  I think I just figured it wasn't really necessary in that we don't need it when we await it here:

https://github.com/tnc-ca-geo/animl-api/blob/d2efca993fd1107776272d06ff5c643435878bcb/src/api/handler.ts#L47

However, I added an export to be more explicit: 4c6a5f792f75ba4cda5744c0f731bf5de1e587e8